extras/rya.console/src/main/java/mvm/rya/shell/command/mongo/MongoConnectionDetails.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package mvm.rya.shell.command.mongo;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * The information that the shell used to connect to Mongo.
+ */
+@Immutable
+@ParametersAreNonnullByDefault
+public class MongoConnectionDetails {
+    private final String username;
+    private final char[] password;
+    private final String instanceName;
+    private final String hostname;
```

Comment:
You can only host multiple instances of Mongo is if they're running on different ports. If we want to support multiple mongo instances (which I don't think we do right now), include a port here instead of instanceName.